mlir/include/mlir/Dialect/Affine/IR/AffineOps.td

```diff
@@ -1153,6 +1153,10 @@ def AffineDelinearizeIndexOp : Affine_Op<"delinearize_index", [Pure]> {
     /// there is no outer bound specified, the leading entry of this result will be
     /// nullptr.
     SmallVector<OpFoldResult> getPaddedBasis();
+
+    /// Returns true if the result of this operation can be used as dimension id
+    /// within 'region', i.e., for all its uses with `region`.
+    bool isValidDim(Region *region);
```

Comment:
This seems like it doesn't belong in this operation. It is some restriction out by ops where this shows up. So the validity should be determined there and not here. Another option is interfaces and external implementation of interfaces

Reply:
This is consistent with what AffineApplyOp does?